tools/system_libs.py

```diff
@@ -1296,15 +1298,25 @@ def get_cflags(self):
       cflags += ['-DMALLOC_FAILURE_ACTION=', '-DEMSCRIPTEN_NO_ERRNO']
     if self.is_tracing:
       cflags += ['--tracing']
+    # Target dlmalloc multithreading aware implementation if using pthreads or Wasm Workers
+    cflags += [f'-DUSE_LOCKS={self.thread_safe}']
+    # When using Wasm Workers, target spinlocks. Otherwise target pthread mutexes when using pthreads.
+    cflags += [f'-DUSE_SPIN_LOCKS={self.use_spinlocks}']
```

Comment:
Why not just always defined USE_SPIN_LOCKS for consistency ... and it looks like USE_LOCKS gets defined to 1 when USE_SPIN_LOCKS is set.

Reply:
If I used USE_SPIN_LOCKS always, the obvious review would be "can you benchmark that it does not regress performance for any user?", which I really don't. I wanted to keep the existing behavior for existing pthreads users to be sure that there is no danger of regressing.

Reply:
> it looks like USE_LOCKS gets defined to 1 when USE_SPIN_LOCKS is set."

I did not want to disrupt the structure of the flags that were set from the way they originally were. The documentation in dlmalloc reads

https://github.com/emscripten-core/emscripten/blob/32831ed6299d8e993cdbc4e7acd7d0ffef5fe2f3/system/lib/dlmalloc.c#L157-L162

suggesting that the intent is that one first sets `USE_LOCKS` to 1 to enable thread safety, and then further adjusts e.g. with `USE_SPIN_LOCKS` to 0 to customize what type of locking to use.

Originally in dlmalloc.c side Emscripten configured both of these flags, so I opted to preserve the same. Passing `-DUSE_SPIN_LOCKS` without defining `-DUSE_LOCKS` does not follow the wording of the docs, and no harm in explicitly passing both?

Reply:
> If I used USE_SPIN_LOCKS always, the obvious review would be "can you benchmark that it does not regress performance for any user?", which I really don't. I wanted to keep the existing behavior for existing pthreads users to be sure that there is no danger of regressing.

I think this is worth a worthwhile thing to do if only to avoid the extra complexity of adding an additional library variant.   It seems highly beneficial to use to use the wasm code in both wasm workers mode and without, no?  And then we can avoid this new `-ww` variant of the library.